### ISSUE TITLE
Different results when validating one single file (or selection of files) against all files in folder

### ISSUE THREAD

**Issue by JozefAerts** [cdisc-org/cdisc-rules-engine#355]
I noticed that when validating single files (using "-dp") or a selection of files gives different results (on files present) than when validating all the files in the folder (using "-d").
It looks as in the first case many rules are skipped that should be executed anyway.
For example: 
python core.py validate -s sdtmig -v 3.4 **-d D:\MetaDataSubmissionGuide_2_0_SASXPTFiles_CORE_testing**  -o d:\temp\report
gives error for vs.xpt with "CORE-000081 - VSSTAT is present when VSPRESP is not present."
When running:
python core.py validate -s sdtmig -v 3.4 **-dp D:\MetaDataSubmissionGuide_2_0_SASXPTFiles_CORE_testing\vs.xpt**  -o d:\temp\report
rule CORE-000081 is skipped or does not give the expected error.
I attach both the report files:
report_all_files_all_rules.xlsx  :  for the first command, i.e. validating all files in the folder
report_VS_only_all_rules.xlsx : for the second command, i.e. only validating vs.xpt
and the XPT files (from the Metadata Submission Guide 2.0)
The strange thing is that CORE-000081 is only applicable to EVENTS, INTERVENTIONS.
So, it looks as when all files in the folder are validated, this is a false positive, and the rule is correctly applied when validation of vs.xpt is done.
I have however also e.g. seen that rule CORE-000219 (--TESTCD > 8 characters) was skipped when only VS was selected for validation.
Will try to select a set of test files for that case.
[MetaDataSubmissionGuide_2_0_SASXPTFiles_CORE_testing.zip](https://github.com/cdisc-org/cdisc-rules-engine/files/11123393/MetaDataSubmissionGuide_2_0_SASXPTFiles_CORE_testing.zip)
[report_VS_only_all_rules.xlsx](https://github.com/cdisc-org/cdisc-rules-engine/files/11123395/report_VS_only_all_rules.xlsx)
[report_all_files_all_rules.xlsx](https://github.com/cdisc-org/cdisc-rules-engine/files/11123396/report_all_files_all_rules.xlsx)

**Comment by OGarcia11** [comment]
Is this still valid? @JozefAerts

**Comment by JozefAerts** [comment]
I need to check ... Will do so in the next few day.

**Comment by JozefAerts** [comment]
Looks like, but in a lesser degree than before. I tested with version 0.9.2.
For example, for dm.xpt from the "Metadata Submission Guide 2.0" files, 
when validating all files in the folder, I get one messages for DM:
"Variable label is not in title case."
When only validating dm.xpt alone, I get two messages:
"At least one required variable is missing from dataset"
"Variable label is not in title case."

For lb.xpt, I did not find any differences.
I will do further extensive checks today and in the next few days.

P.S. The message about variable label is not in title case is for "Reason Arm and/or Actual Arm is Null" which I consider a false positive, which may be related to the slash "/" in it. This is also something we will need to check, but is something different

**Comment by chowsanthony** [comment]
Previous comments mentioned about Metadata Submissions Guide 2.0 files. Those files can be found here: https://www.cdisc.org/standards/foundational/sdtm/sdtm-metadata-submission-guidelines-v2-0#sv__files_links

**Comment by alexfurmenkov** [comment]
Hey @JozefAerts, thanks for providing the data. I think it's already a bit deprecated considering that the issue was raised in 2023.

If I get it right, the problem you highlighted is that when you validate all datasets against all rules the engine flags an issue for the VS dataset, but when you validate the VS dataset only against all rules the engine doesn't flag any issues.

I've reran your .zip against all rules and today's engine version and can report the following results:
- All Datasets vs All Rules: no issues for VS reported
- VS only vs All Rules: 2 issues reported: DM dataset is missing. & EX is not present in study when study includes protocol-specified study treatment

And I think it's OK since looks like the rules CORE-000581 and CORE-000739 require DM and EX to be provided, so after providing these 2 datasets as well there were no issues reported for the VS dataset. Here is the command I've used to validate VS and supply DM and EX:

```
validate
-s
sdtmig
-v
3-4
-dp
MetaDataSubmissionGuide_2_0_SASXPTFiles_CORE_testing/vs.xpt
-dp
MetaDataSubmissionGuide_2_0_SASXPTFiles_CORE_testing/dm.xpt
-dp
MetaDataSubmissionGuide_2_0_SASXPTFiles_CORE_testing/ex.xpt
-dxp
MetaDataSubmissionGuide_2_0_SASXPTFiles_CORE_testing/define.xml
-l
info
```

To sum it up:

- All Datasets vs All Rules: no issues for VS reported
- VS with additionally supplied DM and EX: no issues for VS reported


Can we close the issue or am I missing something?

**Comment by JozefAerts** [comment]
Hi @alexfurmenkov, yes these are to be expected. 
When I run v. 0-14-0 for VS, EX, DM only I also got violations like for EX: "EPOCH is not in TA.EPOCH", but that is to be expected as TA was not in the test set.
I suggest that you close the issue.
I will discuss this with the rules development team, as, i.m.o. such "violations" should not appear in the report when not all datasets in the test set: one cannot test on something that isn't there. But I think many in the team will disagree with me. 
IF the team agrees with my opinion, this should be solved at the YAML level, in the above example by first testing whether TA is present.
Many thanks again for all your work!

**Comment by JozefAerts** [comment]
Hi @alexfurmenkov You can close this issue.
With version 0-14-0, when doing CORE validation on all datasets, I do not get any errors reported for VS, also not for CORE-000081, which is expected, as this rule only applies to "Events" and "Interventions" (VS is "Findings").
When I only do validation on VS (i.e. "VS only") I get some errors for VS, but they are all related to the absence of other datasets/domains, so everything as expected:

<img width="905" height="231" alt="Image" src="https://github.com/user-attachments/assets/37777580-a9b1-4f1b-9555-376294c3d8ca" />

So, yes, you can close the issue.
If, in my daily work, I still find something similar, we can reopen.
